server/src/main/java/org/opensearch/cluster/routing/remote/RemoteRoutingTableService.java

```diff
@@ -0,0 +1,223 @@
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+/**
+ * A Service which provides APIs to upload and download routing table from remote store.
+ *
+ * @opensearch.internal
+ */
+public class RemoteRoutingTableService implements Closeable {
+
+    /**
+     * Cluster setting to specify if routing table should be published to remote store
+     */
+    public static final Setting<Boolean> REMOTE_ROUTING_TABLE_ENABLED_SETTING = Setting.boolSetting(
+        "cluster.remote_store.routing.enabled",
+        true,
+        Setting.Property.NodeScope,
+        Setting.Property.Final
+    );
+    public static final String INDEX_ROUTING_PATH_TOKEN = "index-routing";
+    public static final String ROUTING_TABLE = "routing-table";
+    public static final String INDEX_ROUTING_FILE_PREFIX = "index_routing";
+    public static final String DELIMITER = "__";
+    public static final String INDEX_ROUTING_METADATA_PREFIX = "indexRouting--";
+    private static final Logger logger = LogManager.getLogger(RemoteRoutingTableService.class);
+    private final Settings settings;
+    private final Supplier<RepositoriesService> repositoriesService;
+    private BlobStoreRepository blobStoreRepository;
+    private final ThreadPool threadPool;
+
+    public RemoteRoutingTableService(Supplier<RepositoriesService> repositoriesService,
+                                     Settings settings,
+                                     ThreadPool threadPool) {
+        assert isRemoteRoutingTableEnabled(settings) : "Remote routing table is not enabled";
+        this.repositoriesService = repositoriesService;
+        this.settings = settings;
+        this.threadPool = threadPool;
+    }
+
+    public List<IndexRoutingTable> getChangedIndicesRouting( ClusterState previousClusterState,
+                                                             ClusterState clusterState) {
+        Map<String, IndexRoutingTable> previousIndexRoutingTable = previousClusterState.getRoutingTable().getIndicesRouting();
+        List<IndexRoutingTable> changedIndicesRouting = new ArrayList<>();
+        for (IndexRoutingTable indexRouting : clusterState.getRoutingTable().getIndicesRouting().values()) {
+            if (!(previousIndexRoutingTable.containsKey(indexRouting.getIndex().getName()) && indexRouting.equals(previousIndexRoutingTable.get(indexRouting.getIndex().getName())))) {
+                changedIndicesRouting.add(indexRouting);
+                logger.info("changedIndicesRouting {}", indexRouting.prettyPrint());
+            }
+        }
+
+        return changedIndicesRouting;
+    }
+
+    private String getIndexRoutingFileName() {
+        return String.join(
+            DELIMITER,
+            INDEX_ROUTING_FILE_PREFIX,
+            RemoteStoreUtils.invertLong(System.currentTimeMillis())
+        );
+
+    }
+
+    public CheckedRunnable<IOException> getAsyncIndexMetadataReadAction(
```

Comment:
Where can I see the consumer of this function?

Reply:
https://github.com/shiv0408/OpenSearch/blob/bb335f6da175657a9ef082e3a5f44c9ee3c0d13b/server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java#L1024